build_cmake/helpers.cmake

```diff
@@ -420,17 +420,33 @@ function(config_lib config_name description)
     # Check that the configs dependencies are enabled before setting it to a visible enabled state.
     eval_dependency("${CONFIG_LIB_DEPENDS}" enabled)
     if(enabled)
+        message("-- Looking for library ${CONFIG_LIB_LIB}")
         # 'check_library_exists' won't use our current cache when test compiling the library.
         # To get around this we need to ensure we manually forward WT_ARCH and WT_OS as a minimum. This is particularly
         # needed if 'check_library_exists' will leverage one of our toolchain files.
         if((NOT "${WT_ARCH}" STREQUAL "") AND (NOT "${WT_ARCH}" STREQUAL ""))
             set(CMAKE_REQUIRED_FLAGS "-DWT_ARCH=${WT_ARCH} -DWT_OS=${WT_OS}")
         endif()
-        check_library_exists(${CONFIG_LIB_LIB} ${CONFIG_LIB_FUNC} "" has_lib_${config_name})
```

Comment:
I quite like this change! I think its a much better solution than using the `check_library_exists` macro. 

Though, given we are removing this line we can also probably clean up usage of  `${CONFIG_LIB_FUNC}` (as it was only required by  `check_library_exists`). Would it be possible to remove the `if ("${CONFIG_LIB_FUNC}" STREQUAL "")` check just above this (lines 415-418)?

Reply:
See other reply above, but yes, if you think it won't be wanted again in the future I can get rid of CONFIG_LIB_FUNC and the FUNC arguments to config_lib.